app/code/core/Mage/Sales/Model/Order.php

```diff
@@ -751,7 +751,8 @@ public function canCreditmemo()
          * for this we have additional diapason for 0
          * TotalPaid - contains amount, that were not rounded.
          */
-        if (abs($this->getStore()->roundPrice($this->getTotalPaid()) - $this->getTotalRefunded()) < .0001) {
+        if (abs($this->getStore()->roundPrice($this->getTotalPaid()) - $this->getTotalRefunded()) < .0001
+            && $this->getPayment()->getMethod() !== 'free') {
```

Comment:
This line causes the problem highlighted by @elidrissidev. I think it should be remove and the previous line should be changed. Instead of checking for TotalPaid and TotalRefunded we should count how many non-refunded items are still in the sales_flat_order_item, what do you think?